clash-lib/src/Clash/Data/UniqMap.hs

```diff
@@ -0,0 +1,234 @@
+{-
+Copyright   : (C) 2022 Alexander McKenna
```

Comment:
Most of the definitions in this file are copied from existing modules right? In that case, mention the original copyrights too.

Reply:
To be pedantic I have, because the original module had no copyright notice... I'll approximate one by looking at the history for the file